app/controllers/api/base_controller/generic.rb

```diff
@@ -253,9 +251,10 @@ def model_ident(model, type = nil)
         end
       end
 
-      def validate_id(id, type, klass)
-        return if collection_config.resource_identifier(type) != "id"
-        raise NotFoundError, "Invalid #{klass} id #{id} specified" unless id.kind_of?(Integer) || id =~ /\A\d+\z/
+      def validate_id(id, key_id, klass)
+        if key_id == "id" && !id&.integer? || id.nil?
```

Comment:
This is very confusing mixin && and || without the precendence...perhaps:

```suggestion
        if key_id == "id" && (id.nil? || !id.integer?)
```

Reply:
the original code path went through with id if they specified nothing.
So it was not as lenient, and it properly stated that they did not have an id.

If we remove the 'id.nil?` then , then id.nil? case will go through with a name.
I did modify the validation to say that name requires a value as well, so the error should be thrown, but I wanted the text.

I'll try adding parens or something and see if it becomes more clear.